foo_discord_rich/discord/uploader.h

```diff
@@ -0,0 +1,31 @@
+﻿#pragma once
+
+namespace drp::uploader
+{
+
+
+class threaded_process_artwork_uploader : public threaded_process_callback
+{
+public:
+    threaded_process_artwork_uploader( const pfc::map_t<metadb_index_hash, metadb_handle_ptr>& hashes );
+    void on_init(ctx_t p_wnd) override;
+    void run(threaded_process_status & p_status,abort_callback & p_abort) override;
+    void on_done(ctx_t p_wnd,bool p_was_aborted) override;
+private:
+    pfc::map_t<metadb_index_hash, metadb_handle_ptr> hashes_{};
+};
+
+
+struct artwork_info
+{
+    album_art_data_ptr data;
+    const char* path{};
```

Comment:
Just wanted to document this here but on rare occasions the value of this would change during upload. The two times where I had enough logging to find the exact spot, it was in `check_artwork_hash` right after the `artwork_hash` property had been set. This would make the cover art path invalid and thus make the upload fail. Not sure of the exact reason this happens. Using `pfc::string8` instead of char* seemed to fix the problem.